app/controllers/api/v2/registration_controller.rb

```diff
@@ -121,6 +122,19 @@ def prepare_host
         @host.save!
       end
 
+      def associate_vm
+        if @host.compute_resource
+          associator = ComputeResourceHostAssociator.new(@host.compute_resource)
+          associator.associate_host(@host)
+        else
```

Comment:
The compute resource is set from hostgroup. We should not loop throw all available compute resources.